### PR TITLE
Fix entitlement grant funnel monitoring

### PR DESCRIPTION
**Pull request by Yeusepe**
## Summary
- Consolidates the four divergent entitlement insert paths into the single grantEntitlement funnel.
- Removes the sourceProvider ?? 'gumroad' fallback so callers must pass an explicit provider.
- Removes the Lemon Squeezy bespoke insert bypass.
- Adds a structured per-grant monitoring event.
- Logs a reason for skipped/no-match grant cases that were previously silent.

## Verification
- bun run test:convex

## Behavior
Behavior-preserving: all existing grant tests stay green, and the orchestrator already verified the real-backend E2E provider-parity matrix with the correct sourceProvider per provider.

## Follow-ups
- Reconciliation job.
- Grant-rate anomaly alert.
- Synthetic per-provider canary.

## Summary by CodeRabbit

- **New Features**
  - Centralized, idempotent entitlement granting across purchase, license verification, and webhook flows with consistent evidence upserts and role-sync/audit emission.
  - Structured entitlement grant logging now clearly reports skipped vs error outcomes using hashed identifier metadata.
- **Bug Fixes**
  - Improved Payhip behavior when no catalog tier/product match exists, ensuring correct provider attribution and webhook-derived grant timing.
  - Ensured consistent reactivation and evidence handling during license verification.
- **Tests**
  - Added real/integration coverage for idempotency+logging and Payhip webhook edge cases.
  - Updated mocks/spies reset in entitlement lifecycle tests.

### CODE REVIEW

**Comment by Yeusepe** [comment]
@codex review

File: convex/entitlements.ts [hunk 1]
```diff
@@ -179,6 +279,502 @@ async function requireActiveSubject(
...
+  const metadata = {
+    event: 'entitlement_grant',
+    provider: params.provider,
+    outcome: params.outcome,
+    reason: params.reason,
+    authUserId: params.authUserId,
+    subjectId: params.subjectId,
+    productId: params.productId,
+    entitlementId: params.entitlementId,
+  };
```
[the remark this answers is not in the document]
Comment (Yeusepe) [reply convex/entitlements.ts:303]: Fixed in bf3ac735. Identifier fields now log as sha256Hex hashes (`authUserIdHash`, `subjectIdHash`, `productIdHash`) while provider, outcome, reason, and entitlementId remain plaintext.

File: convex/licenseVerification.ts [hunk 2]
```diff
@@ -502,170 +445,74 @@ export const completeLicenseVerification = mutation({
...
     for (const product of args.productsToGrant) {
-      // Check for existing entitlement (idempotency)
-      const existingEntitlement = await findExistingEntitlementForGrant(ctx, {
+      const existingEntitlements = await ctx.db
+        .query('entitlements')
+        .withIndex('by_auth_user_subject', (q) =>
+          q.eq('authUserId', creatorAuthUserId).eq('subjectId', buyerSubjectId)
+        )
+        .collect();
+      const policySnapshotVersion = existingEntitlements.length + 1;
```
[the remark this answers is not in the document]
Comment (Yeusepe) [reply convex/licenseVerification.ts]: Fixed in bf3ac735. The entitlement collect now runs once before the loop, and `nextPolicySnapshotVersion` advances only when the funnel inserts a new entitlement.

File: convex/entitlements.ts [hunk 3]
```diff
@@ -179,6 +280,502 @@ async function requireActiveSubject(
...
+      previousStatus: undefined,
+      outboxJobIds,
+      evidenceId,
+    };
+  } catch (err) {
+    const reason = err instanceof ConvexError ? String(err.data) : err instanceof Error ? err.message : 'error';
+    await logEntitlementGrantAttempt({
+      provider,
+      outcome: 'error',
+      reason,
```
[the remark this answers is not in the document]
Comment (Yeusepe) [reply convex/entitlements.ts]: Fixed in edc3ef76. Unexpected grant errors now log the bounded `unexpected_error` reason while known string ConvexError reasons stay explicit, and the regression covers the stale-subject path.
